components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Response.kt

```diff
@@ -64,16 +82,16 @@ data class Response(
          * Executes the given [block] function with the buffered reader as parameter and then closes it down correctly
          * whether an exception is thrown or not.
          */
-        fun <R> useBufferedReader(charset: Charset = Charsets.UTF_8, block: (BufferedReader) -> R): R = use {
-            block(stream.bufferedReader(charset))
+        fun <R> useBufferedReader(charset: Charset? = null, block: (BufferedReader) -> R): R = use {
+            block(stream.bufferedReader(charset ?: this.charset))
         }
 
         /**
          * Reads this body completely as a String.
          *
          * Takes care of closing the body down correctly whether an exception is thrown or not.
          */
-        fun string(charset: Charset = Charsets.UTF_8): String = useBufferedReader(charset) { it.readText() }
```

Comment:
Should we allow the charset override here too?

I'm going to merge now; we can do that in a follow-up if we want to. :)

Reply:
@pocmo OK, I will send a follow-up PR.